jerry-core/vm/vm.c

```diff
@@ -3307,6 +3304,31 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           ecma_fast_free_value (value);
           continue;
         }
+        case VM_OC_BRANCH_OPTIONAL_CHAIN:
+        {
+          left_value = stack_top_p[-1];
+
+          bool pop_reference = byte_code_p[0] == CBC_EXT_OPCODE && byte_code_p[1] == CBC_EXT_POP_REFERENCE;
+
+          if (!ecma_is_value_null (left_value) && !ecma_is_value_undefined (left_value))
```

Comment:
Please lets this be: `if (!(ecma_is_value_null (left_value) || ecma_is_value_undefined (left_value)))`